src/Common/src/Interop/Windows/mincore/Interop.CreateFile.cs

```diff
@@ -18,5 +19,18 @@ internal partial class mincore
             System.IO.FileMode dwCreationDisposition,
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile);
+
+        internal static SafeFileHandle CreateFile(
```

Comment:
GetFinalPathNameByHandle doesn't seem to care about the original format of the string. I'll spend more time to validate all of the various handle scenarios, but I'm pretty sure that Windows doesn't associate the passed in string with handles.